httpx/_utils.py

```diff
@@ -67,6 +68,8 @@ def primitive_value_to_str(value: "PrimitiveData") -> str:
         return "false"
     elif value is None:
         return ""
+    elif isinstance(value, enum.Enum):
+        return str(value.value)
```

Comment:
Does this imply that the `PrimitiveData` type ought to change to include `enum.Enum`?

Reply:
Yeah, I miss that.

Do you think that `enum.Enum` should not be considered a primitive data, and hence not be in this function?

Reply:
My honest answer?
My preference would be for us to raise nice tight `TypeError` cases here.

Reply:
For a similar example...

```python
import json

json.dumps{"value": Color.RED}        # TypeError: Object of type Color is not JSON serializable
json.dumps{"value": Color.RED.value}  # '{"value": 1}'
```

Reply:
The thing is... I think the following should raise an error:
```python
class Color(enum.Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
```

But not the following:

```python
class Color(int, enum.Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
```
Because the `int` there means that each element is an integer, analogous to other types.

Reply:
Hrm. That's a bit awkward...

```python
class Color(str, enum.Enum):
    # heeeeyyyyyy I'm just a normal str
    RED = "red"
    GREEN = "green"
    BLUE = "blue"

c = Color.RED
isinstance(c, str)  # True
str(c)              # 'Color.RED'  yeah but you ain't
```

The upshot is *even if you're type-checking* the user still needs to correctly pass `Color.RED.value`, unless we special-case for `enum`.

Maybe the upshot is we should have strict type checking here *and* that we should also expand the primitive types to include `enum.Enum`. (Just because we're friendly like that?)

Reply:
> The upshot is even if you're type-checking the user still needs to correctly pass Color.RED.value, unless we special-case for enum.

If I understood correctly what you said, this is not the case. Type checkers understand that each member of a (str, Enum) is a str, and they don't raise errors.

Reply:
I don't think requests has the same behavior as httpx here, does it? Because the Testclient changed, so I assume requests can do the "right" thing on (str, Enum) - I'm on my phone.